dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoJedisClient.java

```diff
@@ -793,7 +793,11 @@ private String getCursorValue(final ConnectionContext state, final CursorBasedRe
 
     private List<OperationResult<ScanResult<String>>> scatterGatherScan(final CursorBasedResult<String> cursor,
             final int count, final String... pattern) {
-        return new ArrayList<>(connPool.executeWithRing(new BaseKeyOperation<ScanResult<String>>("SCAN", OpName.SCAN) {
+
+        if (!(cursor instanceof TokenRackMapper)) {
```

Comment:
When will this throw an exception? Is there a case that `cursor` is not an instance of `TokenRacMapper?

Reply:
Ideally the cursor is something that we create and send it out. so we do not expect user the create it. Apparently the API isnt good